packages/core-app-api/config.d.ts

```diff
@@ -65,6 +65,21 @@ export interface Config {
         }>;
       }>;
     };
+    /**
+     * A list of application feature flags
+     */
+    featureFlags?: Array<{
```

Comment:
I'm a little torn on this. I'm would adding them to the `createApp` call like we have `createPlugin` work instead of loading these from config? Being able to define them like:

```ts
const app = createApp({
  apis,
  plugins: Object.values(plugins),
  featureFlags: [...],
  icons: {
    // Custom icon example
    alert: AlarmIcon,
  },
  bindRoutes({ bind }) {
      ...
  },
});
```

Does this work for you?

I'm thinking that it would probably make sense for maybe default values to be loaded in from config, but maybe not the flags themselves as they're always tied to code everywhere they are used right now.

Reply:
It's funny because originally it was like this but got feedback from @djamaile to change it to the config file instead. I mean, both work for me but not sure what would be the big picture going forwards?

Reply:
Ah yeh, sorry @djamaile, I think it makes more sense to just have them removed from config for now. If you wanted to still store them in config, I would recommend creating your own implementation of the `FeatureFlagsApi` and overriding the implementation that is used at the app level something like [this](https://backstage.io/docs/api/utility-apis#custom-implementations-of-utility-apis).

Thinking that it kind of keeps it cleanish right now at least and if we want to change into this at a later date then we've got room to do it later.

Reply:
Ok, moved it back to the createApp now

Reply:
@drodil originally it was in the `user-settings` plugin, in the config file. At least thats what I reviewed 😓 Might have missed something though. This looks nice to me!